### PR TITLE
users login and validation

### PR DESCRIPTION
**Pull request by NicholasAJ**
users has been tested and completed. moving to reviews

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -2,12 +2,22 @@ const express = require('express');
...
+// const bodyparser = require('body-parser');
+// app.use(bodyparser.json);
+
+// const user = require('./route/user');
+// app.use('/api/user', user);
+
+// const bcrypt = require('bcrypt');
+// const saltRounds = 10;
+// const myPlaintextPassword = 's0/\/\P4$$w0rD';
+// const someOtherPlaintextPassword = 'not_bacon';
```
Comment (ATDaniel) [inline server/server.js:14]: removed unused comments

File: server/routes/review.routes.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const ReviewController = require('../controllers/review.controller');
+
+module.exports = (app) => {
+  app.put('/review/create', ReviewController.createReview);
+  app.get('/reviews/getall', ReviewController.getAllReviews);
+  app.get('/review/getone', ReviewController.getOneReview);
+  app.put('/review/update', ReviewController.updateReview);
+  app.delete('/review/delete', ReviewController.deleteReview);
```
Comment (ATDaniel) [inline server/routes/review.routes.js:8]: You don't need the `/create` and `/getall` for endpoint routes. That's taken care of in the route's method. They should all share the same route, and just call different functions based on the route method. The exception is the "get one" route, where you need the query parameter, so it'd be `/review/:reviewId`.

File: server/controllers/user.controller.js [hunk 3]
```diff
@@ -48,9 +67,33 @@ const deleteUser = (req, res) => {
...
+const loginUser = (req,res) => {
+  const {username, password} = req.body;
+  User.findOne({username: username}).then(user =>{
+    if(!user) {
+      return res.status(500).json({
+        message: "User does not exist."
+      })
+    }
+    const passwordMatch = bcrypt.compareSync(password, user.password)
+    console.log(passwordMatch);
```
Comment (ATDaniel) [inline server/controllers/user.controller.js:79]: You can remove this log now that it's working.

File: server/server.js (hunk 1 above)
Comment (NicholasAJ) [reply server/server.js:14]: I have removed the comments

**Review by ATDaniel — APPROVED** [review]

**Pull request merged**